packages/kbn-unified-data-table/src/components/data_table.tsx

```diff
@@ -336,17 +380,43 @@ export const DiscoverGrid = ({
     [displayedRows, dataView, fieldFormats]
   );
 
+  const unifiedDataTableContextValue = useMemo(
+    () => ({
+      expanded: expandedDoc,
+      setExpanded: setExpandedDoc,
+      rows: displayedRows,
+      onFilter,
+      dataView,
+      isDarkMode: darkMode,
+      selectedDocs: usedSelectedDocs,
+      setSelectedDocs: (newSelectedDocs: React.SetStateAction<string[]>) => {
+        setSelectedDocs(newSelectedDocs);
+        if (isFilterActive && newSelectedDocs.length === 0) {
+          setIsFilterActive(false);
+        }
+      },
+      valueToStringConverter,
+    }),
+    [
+      darkMode,
+      dataView,
+      displayedRows,
+      expandedDoc,
+      isFilterActive,
+      onFilter,
+      setExpandedDoc,
+      usedSelectedDocs,
+      valueToStringConverter,
+    ]
+  );
+
   /**
    * Pagination
    */
-  const defaultRowsPerPage = useMemo(
-    () => getDefaultRowsPerPage(services.uiSettings),
-    [services.uiSettings]
-  );
   const currentPageSize =
     typeof rowsPerPageState === 'number' && rowsPerPageState > 0
       ? rowsPerPageState
-      : defaultRowsPerPage;
+      : DEFAULT_ROWS_PER_PAGE;
   const [pagination, setPagination] = useState({
     pageIndex: 0,
```

Comment:
Is there a reason pageIndex is handled by internal state? I was surprised to see it not in the list of props. Is this a design decision wrt to browser history maybe?

Reply:
It's just that there was no need to handle the pagination externally in Discover so far, which doesn't mean it could be useful for other use cases

Reply:
Cool, yea I'm looking to make use of this new table in Cloud security (Findings). Currently pageIndex is part of our url history, but I wonder what the standard across all of kibana is. I can't imagine deep linking to a particular page of results is all that useful, but the back/forward behavior wrt to switching pages could be something customers expect. cc @dimadavid if you had any thoughts around this, and if we have any established UX guidelines. cc @kfirpeled (is it important we keep pageIndex as part of the urlQuery?)